src/main/kotlin/com/faforever/userservice/backend/tos/TosService.kt

```diff
@@ -0,0 +1,35 @@
+package com.faforever.userservice.backend.tos
+
+import com.faforever.userservice.backend.domain.TermsOfService
+import com.faforever.userservice.backend.domain.TermsOfServiceRepository
+import com.faforever.userservice.backend.domain.UserRepository
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.transaction.Transactional
+
+
+@ApplicationScoped
+class TosService(
+    private val userRepository: UserRepository,
+    private val tosRepository: TermsOfServiceRepository,
+) {
+    fun hasUserAcceptedLatestTos(userId: Int): Boolean {
+        val user = userRepository.findById(userId)
+        user?.let {
+            val latestTos = tosRepository.findLatest()
+            return latestTos?.version == user.acceptedTos
+        }
+        return false
+    }
+
+    fun findLatestTos(): TermsOfService? = tosRepository.findLatest()
+
+    @Transactional
+    fun acceptLatestTos(userId: Int) {
+        val user = userRepository.findById(userId)
+        user?.let {
+            val latestTos = tosRepository.findLatest()
+            user.acceptedTos = latestTos?.version
+            userRepository.persist(user)
+        }
```

Comment:
A null user should rather throw an exception instead of doing nothing

```
        val user = userRepository.findById(userId)
            ?: throw IllegalStateException("User id $userId not found")

       val latestTos = tosRepository.findLatest()
       user.acceptedTos = latestTos?.version
       userRepository.persist(user)
```